frontend/src/DialectReportView.tsx

```diff
@@ -22,27 +28,82 @@ export const DialectReportView = ({
-    if (params.getAll("language").length) {
-      const filteredArray = Array.from(
+    const filteredReportData = { ...reportData };
+    let filteredOtherImplementationsData: Record<string, Implementation> = {};
+    const selectedLanguages = params.getAll("language");
+
+    if (selectedLanguages.length > 0) {
+      const filteredReportArray = Array.from(
         reportData.implementations.entries(),
       ).filter(([, data]) =>
-        params.getAll("language").includes(data.metadata.language),
+        selectedLanguages.includes(data.metadata.language),
+      );
+      const filteredReportImplementationsMap = new Map(filteredReportArray);
+      filteredReportData.implementations = filteredReportImplementationsMap;
+      if (allImplementationsData) {
+        filteredOtherImplementationsData = filterOtherImplementations(
+          allImplementationsData,
+          selectedLanguages,
+          filteredReportImplementationsMap,
+        );
+      }
+    } else {
+      const filteredReportArray = Array.from(
+        reportData.implementations.entries(),
       );
-      filteredData = { ...reportData, implementations: new Map(filteredArray) };
+      const filteredReportImplementationsMap = new Map(filteredReportArray);
+      if (allImplementationsData) {
+        filteredOtherImplementationsData = filterOtherImplementations(
+          allImplementationsData,
+          languages,
+          filteredReportImplementationsMap,
+        );
+      }
     }
-    return filteredData;
-  }, [reportData, params]);
+    return { filteredReportData, filteredOtherImplementationsData };
+  }, [reportData, params, allImplementationsData, languages]);
 
   return (
     <div>
       <div className="container p-4">
         <BowtieInfoSection />
-        <RunInfoSection runInfo={filteredData.runInfo} />
+        <RunInfoSection runInfo={filteredData.filteredReportData.runInfo} />
         <FilterSection languages={languages} />
-        <SummarySection reportData={filteredData} />
-        <CasesSection reportData={filteredData} />
+        <SummarySection
+          reportData={filteredData.filteredReportData}
+          otherImplementationsData={
+            filteredData.filteredOtherImplementationsData
+          }
+        />
+        <CasesSection reportData={filteredData.filteredReportData} />
       </div>
     </div>
   );
 };
+
+const filterOtherImplementations = (
+  allImplementationsData: Record<string, Implementation>,
+  langs: string[],
+  filteredReportImplementationsMap: Map<string, ImplementationData>,
+): Record<string, Implementation> => {
+  const filteredOtherImplementationsArray: [string, Implementation][] =
+    Object.entries(allImplementationsData).filter(
+      ([, impl]: [string, Implementation]) => langs.includes(impl.language),
+    );
+
+  const filteredOtherImplementationsData: Record<string, Implementation> = {};
+  filteredOtherImplementationsArray.forEach(([key, impl]) => {
```

Comment:
This forEach basically transforms this collection - using `map()` should be more clear